docs/reference/classes.md

```diff
@@ -309,3 +309,37 @@ inside that class.
 Even more specifically, if you declare a [companion object](object-declarations.html#companion-objects) inside your class,
 you'll be able to call its members with the same syntax as calling static methods in Java/C#, using only the class name
 as a qualifier.
+
+
+## Sealed Classes
+
+Sealed classes are used for representing restricted class hierarchies, when a value can have one of the types from a
+limited set, but cannot have any other type. They are, in a sense, an extension of enum classes: the set of values
```

Comment:
Comparing to enum classes is a very long shot, to my mind. Referring to algebraic data types explicitly would be better.